### ISSUE TITLE
pipelines can hang if job stuck queueing for too long with no nodes available

### ISSUE THREAD

**Issue by andrew-m-leonard** [adoptium/ci-jenkins-pipelines#677]
Consider having a timeout for waiting for a node when "none" are available. Distinguish this to none available, due to them all being busy, so none here means "none at all"

**Comment by adamfarley** [comment]
I agree. Perhaps the job status could be "aborted" due to timeout, and "failed" if no nodes exist?
